app/src/test/java/com/samco/trackandgraph/statistics/Statistics_getYParameters_KtTest.kt

```diff
@@ -0,0 +1,312 @@
+    fun manuallyTestValuesTime() {
+        runBlocking {
+            val y_min = 1 *60*60 + 53 *60 // in seconds
+            val y_max = 8 *60*60 + 25 *60
+
+            printExampleTime(y_min.toDouble(), y_max.toDouble())
+        }
+    }
+
+    @Test
+    @ExperimentalTime
+    fun printExamplesTime() {
+        val RANGE_USED_BELOW = 0.82
+        for (q_ in 0 until 32) {
+
+            for (qq_ in 0 until 200) {
+                val start = Random.nextInt(0, 3 * 60 * 60).toDouble()
+                val length = Random.nextInt(1, 10 * 60 * 60).toDouble()
+                val end = start + length
+
+                val interval = try {
+                    getYParametersInternal(start, end, time_data = true, fixedBounds = false)
+                } catch (e: Exception) {
+                    PossibleInterval((end - start) / 10.0, false, 11, 1.0,
+                            start, end, 1.0)
+                }
+
+                if (interval.percentage_range_used > RANGE_USED_BELOW) continue
+
+                printExampleTime(start, end)
+                break
+            }
+        }
+    }
+
+    @Test
+    fun printExamplesNumerical() {
+        val RANGE_USED_BELOW = 0.82
+        for (q_ in 0 until 32) {
+
+            for (qq_ in 0 until 200) {
+                // generate new numbers until you get in the range_below regime. 201 tries should be enough.
+                val start = Random.nextInt(0, 1000).toDouble() / 10
+                val length = Random.nextInt(1, 500).toDouble() / 10
+                val end = start + length
+
+                val interval = try {
+                    getYParametersInternal(start, end, time_data = false, fixedBounds = false)
+                } catch (e: Exception) {
+                    PossibleInterval((end - start) / 10.0, false, 11, 1.0,
+                            start, end, 1.0)
+                }
+
+                if (interval.percentage_range_used > RANGE_USED_BELOW) continue
+
+                printExampleNumerical(start, end)
+                break
+            }
+
+        }
+    }
+
+    fun find_solution_for_everything(time_data: Boolean) {
+        var errors = 0
+        val allowed_error_percentage = when (time_data) {
+            false -> 0.1 // it's ok if one   in 1000 doesn't have a good solution
+            true  -> 0.7 // it's ok if seven in 1000 don't   have a good solution
+        }
+        val no_solution_vals = mutableListOf<Pair<Double, Double>>()
+        val intervalList = mutableListOf<PossibleInterval>()
+
+        val startValues  = when (time_data) {
+            false -> (0..10000).map { it.toDouble()/10 }
+            true  -> (0..36000).map { it.toDouble() } // zero seconds to 10 hours
+        }
+        val lengthValues = when (time_data) {
+            false -> (1..7500).map { it.toDouble()/10 }
+            true  -> (1..72000).map { it.toDouble() } // one second to 20 hours
+        }
+
+        for (start in startValues.shuffled().slice(0..10000)) {
```

Comment:
I appreciate knowing that the code is this consistent but I think this test is a little annoyingly slow to run in quick succession. I suggest we reduce iterations on these tests considerably.

Reply:
Yeah, maybe add a flag whether you want to do an excessive test or just a quick check, with the quick one being the default.